hugegraph-core/src/main/java/com/baidu/hugegraph/security/HugeSecurityManager.java

```diff
@@ -51,171 +59,197 @@
     }
 
     @Override
     public void checkRead(String file) {
-        if (callFromGremlin() && !callFromCaffeine()) {
-            throw new SecurityException("Not allowed to read file via Gremlin");
+        if (callFromGremlin() && !callFromCaffeine() &&
+            !readGroovyInCurrentDir(file) && !callFromBackendHbase()) {
+            throw newSecurityException(
+                  "Not allowed to read file via Gremlin: %s", file);
         }
         super.checkRead(file);
     }
 
     @Override
     public void checkRead(String file, Object context) {
         if (callFromGremlin()) {
-            throw new SecurityException("Not allowed to read file via Gremlin");
+            throw newSecurityException(
+                  "Not allowed to read file via Gremlin: %s", file);
         }
         super.checkRead(file, context);
     }
 
     @Override
     public void checkWrite(FileDescriptor fd) {
-        if (callFromGremlin()) {
-            throw new SecurityException("Not allowed to write fd via Gremlin");
+        if (callFromGremlin() && !callFromBackendSocket()) {
+            throw newSecurityException("Not allowed to write fd via Gremlin");
         }
         super.checkWrite(fd);
     }
 
     @Override
     public void checkWrite(String file) {
         if (callFromGremlin()) {
-            throw new SecurityException(
-                      "Not allowed to write file via Gremlin");
+            throw newSecurityException("Not allowed to write file via Gremlin");
         }
         super.checkWrite(file);
     }
 
     @Override
     public void checkDelete(String file) {
         if (callFromGremlin()) {
-            throw new SecurityException(
-                      "Not allowed to delete file via Gremlin");
+            throw newSecurityException(
+                  "Not allowed to delete file via Gremlin");
         }
         super.checkDelete(file);
     }
 
     @Override
     public void checkListen(int port) {
         if (callFromGremlin()) {
-            throw new SecurityException(
-                      "Not allowed to listen socket via Gremlin");
+            throw newSecurityException(
+                  "Not allowed to listen socket via Gremlin");
         }
         super.checkListen(port);
     }
 
     @Override
     public void checkAccept(String host, int port) {
         if (callFromGremlin()) {
-            throw new SecurityException(
-                      "Not allowed to accept socket via Gremlin");
+            throw newSecurityException(
+                  "Not allowed to accept socket via Gremlin");
         }
         super.checkAccept(host, port);
     }
 
     @Override
     public void checkConnect(String host, int port) {
-        if (callFromGremlin()) {
-            throw new SecurityException(
-                      "Not allowed to connect socket via Gremlin");
+        if (callFromGremlin() && !callFromBackendSocket() &&
+            !callFromBackendHbase()) {
```

Comment:
seems don't have to add callFromBackendHbase(), checkRead/checkWrite are ok when no callFromBackendHbase()

Reply:
hbase also connect socket. refer:https://github.com/hugegraph/hugegraph/issues/758#issuecomment-562598720